tests/actions.go

```diff
@@ -1711,6 +1553,173 @@ func (oa *OperatorActions) tidbMembersReadyFn(tc *v1alpha1.TidbCluster) (bool, e
 	return true, nil
 }
 
+func (oa *OperatorActions) dmMasterMembersReadyFn(dc *v1alpha1.DMCluster) bool {
+	dcName := dc.GetName()
+	ns := dc.GetNamespace()
+	masterSetName := controller.DMMasterMemberName(dcName)
+
+	masterSet, err := oa.tcStsGetter.StatefulSets(ns).Get(masterSetName, metav1.GetOptions{})
+	if err != nil {
+		log.Logf("failed to get statefulset: %s/%s, %v", ns, masterSetName, err)
+		return false
+	}
+
+	if masterSet.Status.CurrentRevision != masterSet.Status.UpdateRevision {
+		log.Logf("dm master .Status.CurrentRevision (%s) != .Status.UpdateRevision (%s)", masterSet.Status.CurrentRevision, masterSet.Status.UpdateRevision)
+		return false
+	}
+
+	if !utilstatefulset.IsAllDesiredPodsRunningAndReady(helper.NewHijackClient(oa.kubeCli, oa.asCli), masterSet) {
+		return false
+	}
+
+	if dc.Status.Master.StatefulSet == nil {
+		log.Logf("DmCluster: %s/%s .status.Master.StatefulSet is nil", ns, dcName)
+		return false
+	}
+
+	failureCount := len(dc.Status.Master.FailureMembers)
+	replicas := dc.Spec.Master.Replicas + int32(failureCount)
+	if *masterSet.Spec.Replicas != replicas {
+		log.Logf("statefulset: %s/%s .spec.Replicas(%d) != %d",
+			ns, masterSetName, *masterSet.Spec.Replicas, replicas)
+		return false
+	}
+	if masterSet.Status.ReadyReplicas != dc.Spec.Master.Replicas {
```

Comment:
`dc.Spec.Master.Replicas` does not count the failureCount in.

Reply:
Can we just compare `dc.Spec.Master.Replicas` with `.Status.ReadyReplicas` (ignore failure count) to verify the component is ready?

Reply:
updated to use `dc.MasterAllPodsStarted` and `dc.MasterAllMembersReady`.